File: first_package/src/minimal_demo.cpp
#include <rclcpp/rclcpp.hpp>

int main(int argc, char* argv[]) {
  rclcpp::init(argc, argv);
  auto node = rclcpp::Node::make_shared("minimal_cpp");
  RCLCPP_INFO_STREAM(node->get_logger(), "Hello from " << node->get_name());
  RCLCPP_ERROR_STREAM(node->get_logger(), "Hello from " << node->get_name());
  RCLCPP_FATAL_STREAM(node->get_logger(), "Hello from " << node->get_name());
  RCLCPP_WARN_STREAM(node->get_logger(), "Hello from " << node->get_name());

  rclcpp::spin(node);
  rclcpp::shutdown();
}

File: first_package/script/minimal_demo.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

def main(args=None):
    rclpy.init(args=args)
    node = Node("minimal_py")
    node.get_logger().info(f"Hello from {node.get_name()}")
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        # Log a message when the node is manually terminated
        node.get_logger().warn("Keyboard interrupt detected")
    finally:
        # Shut down the ROS 2 Python client library
        rclpy.shutdown()
    
if __name__ == "__main__":
    main()